### PR TITLE
✨ [Story interactive] Layout and Styles for Detailed Results Component

### PR DESCRIPTION
**Pull request by Brandons42**
Establishes layout of detailed result elements randomly around center image, as well as coloring for correct/incorrect quiz results; not yet suitable for high number of results

Demo: https://interactive-img.web.app/res/layout/

/cc @mszylkowski 
/cc @processprocess

### CODE REVIEW

**Comment by mszylkowski** [comment]
image
The text options should not be transparent on the transparent theme (and probably not at all).

File: examples/amp-story/interactive_results_detailed.html [hunk 1]
```diff
@@ -339,25 +270,93 @@ <h1>Detailed Results Component</h1>
...
+          poll.setAttribute('endpoint', 'https://webstoriesinteractivity-beta.web.app/api/v1');
+          poll.setAttribute('option-1-text', 'Lake Tahoe');
+          poll.setAttribute('option-1-results-category', 'Lake lover');
+          poll.setAttribute('option-2-text', 'Hong Kong');
+          poll.setAttribute('option-2-results-category', 'City-goer');
+          polls.append(poll);
+        }
+      }
+
+      document.getElementById('add-img-polls').onclick = function () {
```
Comment (mszylkowski) [inline examples/amp-story/interactive_results_detailed.html:344]: This logic is very clever but tying these elements to the pages makes the story behave weirdly. Eg: sliding on the range will advance the page. Sth that would make more sense is:
- Add a sibling of amp-story that's positioned in the top left corner
- Contain 4 input ranges that go from 0-5 and set the initial value to 2-3 (the story should start with some components on each page)
- Listen for `change` events on the input ranges and update the number of components. No need for an extra button.

This way, you can use an overlay to modify the story, but by default you can test it. There's a few places where we do this (add an overlay near a corner to modify the story)

File: extensions/amp-story/1.0/amp-story.css [hunk 2]
```diff
@@ -924,7 +924,7 @@ amp-story .amp-video-eq,
   --i-amphtml-interactive-component-shadow: 0px 4px 12px rgba(60, 64, 67, 0.08), 0px 2px 4px rgba(60, 64, 67, 0.12);
   --i-amphtml-interactive-prompt-clamp: 4 !important;
   --i-amphtml-interactive-results-prompt-margin: 20px !important;
-  --i-amphtml-interactive-placeholder-background: rgba(32, 33, 36, 0.2) !important;
+  --i-amphtml-interactive-placeholder-background: rgb(210,211,211) !important;
```
Comment (processprocess) [inline extensions/amp-story/1.0/amp-story.css:927]: nit: please spaces between values :)

**Review by processprocess — COMMENTED** [review]
This is awesome. Please leave comments in the `positionResultEls_` function. Other than that I think it looks good.

File: extensions/amp-story-interactive/0.1/amp-story-interactive-results-detailed.js [hunk 3]
```diff
@@ -186,32 +186,36 @@ export class AmpStoryInteractiveResultsDetailed extends AmpStoryInteractiveResul
...
-      const top = CENTER + Math.cos(angle) * dist - size / 2;
-      const left = CENTER + Math.sin(angle) * dist - size / 2;
+      // Angle to place the result at relative to its slice; [angleBuffer, slice - angleBuffer)
+      const sliceRelAngle =
+        Math.random() * (slice - 2 * angleBuffer) + angleBuffer;
+      const angle = slice * index + sliceRelAngle + offset;
+
+      // Transform the angle and distance (polar coordinates) of the result
+      // into Cartesian coordinates to be used in CSS
+      const transform = (trig) => CENTER + trig(angle) * dist - size / 2 + 'em';
```
Comment (processprocess) [inline extensions/amp-story-interactive/0.1/amp-story-interactive-results-detailed.js:213]: This is pretty cool! Please comment that `trig` is either `Math.cos` or `Math.sin`.

**Pull request closed without merging**